v0/destinations/ProfitWell/transform.js

```diff
@@ -0,0 +1,186 @@
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  getFieldValueFromMessage
+} = require("../../util");
+const { ConfigCategory, mappingConfig, baseEndpoint } = require("./config");
+
+const identifyResponseBuilder = async (message, { Config }) => {
+  const response = defaultRequestConfig();
+
+  const userId = getDestinationExternalID(message, "profitwellUserId");
+  const userAlias = getFieldValueFromMessage(message, "userId");
+
+  if (!userId && !userAlias) {
+    throw new CustomError("User not found", 400);
+  }
+
+  let subscriptionId = getDestinationExternalID(
+    message,
+    "profitwellSubscriptionId"
+  );
+  let subscriptionAlias =
+    get(message, "traits.subscriptionAlias") ||
+    get(message, "context.traits.subscriptionAlias");
+
+  const targetUrl = `${baseEndpoint}/v2/users/${userId || userAlias}`;
+  const res = await getSubscriptionHistory(targetUrl, {
+    headers: {
+      Authorization: Config.privateApiKey
+    }
+  });
+
+  let payload;
+  if (res.success) {
+    // some() breaks if the callback returns true
+    let subscriptionFound = true;
+    const valFound = res.response.some(element => {
+      if (userId === element.user_id || userAlias === element.user_alias) {
+        if (subscriptionId === element.subscription_id) {
+          subscriptionId = element.subscription_id;
+          subscriptionFound = true;
+          return true;
+        }
+        if (subscriptionAlias === element.subscription_alias) {
+          subscriptionAlias = element.subscription_alias;
+          subscriptionFound = true;
+          return true;
+        }
+        subscriptionFound = false;
+      }
+      return false;
+    });
+
+    // for a given userId, subscriptionId not found
+    if (!subscriptionFound) {
+      payload = constructPayload(
+        message,
+        mappingConfig[ConfigCategory.IDENTIFY_CREATE.name]
+      );
+      payload = {
+        ...payload,
+        user_id: userId,
+        user_alias: userAlias
+      };
+      payload = removeUndefinedAndNullValues(payload);
+      response.method = defaultPostRequestConfig.requestMethod;
+      response.endpoint = `${baseEndpoint}/v2/subscriptions`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = payload;
+      return response;
+    }
+
+    // userId and SubscriptionId is found
+    if (valFound) {
+      payload = constructPayload(
+        message,
+        mappingConfig[ConfigCategory.IDENTIFY_UPDATE.name]
+      );
+      payload = removeUndefinedAndNullValues(payload);
+      response.method = defaultPutRequestConfig.requestMethod;
+      response.endpoint = `${baseEndpoint}/v2/subscriptions/${subscriptionId ||
+        subscriptionAlias}`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = payload;
+      return response;
+    }
+  }
```

Comment:
we should put a check here for the case where response.success is `false` and status code is not 404. For eg, status is 400, in that case we need to throw a `CustomError`